src/plugins/ipt/ipt.h

```diff
@@ -0,0 +1,174 @@
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef IPT_H
+#define IPT_H
+
+#include <vector>
+#include <memory>
+
+#include <glib.h>
+#include "plugins/private.h"
+#include "plugins/plugins_ex.h"
+
+#define PTW_CURRENT_CR3  (0xC3000000)
+#define PTW_CURRENT_TID  (0x1D000000)
+#define PTW_EVENT_ID     (0xCC000000)
+#define PTW_ERROR_EMPTY  (0xBAD10000)
+
+#define IPT_MAX_VCPUS 16
+
+
+event_response_t ipt_cr3_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
+event_response_t ipt_catchall_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
+
+
+struct ipt_config
+{
+    const char* ipt_dir;
+};
+
+struct ipt_vcpu
+{
+    // pointer to IPT buffer for this vCPU
+    uint8_t* buf;
+
+    // IPT buffer size
+    uint64_t size;
+
+    // where IPT data stream is saved
+    FILE* fd;
+};
+
+class ipt: public pluginex
+{
+    unsigned int num_vcpus;
+
+    drakvuf_trap_t cr3_trap =
+    {
+        .type = REGISTER,
+        .reg = CR3,
+        .data = this,
+        .name = "ipt_cr3_cb",
+        .cb = ipt_cr3_cb
+    };
+
+    drakvuf_trap_t bp_trap =
+    {
+        .type = CATCHALL_BREAKPOINT,
+        .data = this,
+        .name = "ipt_catchall_cb",
+        .cb = ipt_catchall_cb
+    };
+
+public:
+    struct ipt_vcpu vcpu[IPT_MAX_VCPUS];
```

Comment:
```suggestion
    std::array<ipt_vcpu, IPT_MAX_VCPUS> vcpus;
```
Well, you've made this a `.hpp` file, so.....

(this requires `#include <array>`